### DISCUSSION TITLE
Why properties are used

### DISCUSSION THREAD

**Discussion by arteny** [SebastianMC/obsidian-custom-sort#106] [Ideas]
If only way to configure sorting is to use specific file, why you are using properties (metadata) there to configure? It might be much more easy to use just regular lines. Instead, now for every created note I see "sporting-spec" property name auto-suggestion, so I should remember what it means, when actually it is using only for one type of notes (sortspect).

**Comment by SebastianMC** [comment]
Hi @arteny

I get your point.
The plugin is built on top of obsidian internal mechanisms, relies fully and heavily on them. It is intentionally kept lightweight with minimal overhead.
In particular:
- intentionally, no parsing of notes, no hook up to notes changing events, clean and simple
- instead, it reads directly the obsidian metadata cache. Obsidian itself keeps it up to date automatically, (re)reads changed notes as needed. Quick and efficient, no overhead

Then, sorting specifications can be placed in several locations:
- in notes named 'sortspec', the obvious default behavior. There can be many such notes
- in folder notes. In other words, notes with the same name as their parent folder.
  - I'm using this method in my vaults: no additional notes needed, sorting specification is invisible (sits in metadata) and doesn't pollute the folder note body. And the specification is simpler, no need to `target-folder:` because the folder note location self-determined the context
- in note(s) with name (or path) specified by the user in configuration

Another rationale behind reading from a dedicated metadata instead of note(s) body text is that the sorting spec syntax can interfere with the syntax of Obsidian notes, especially plugins. Given that on one hand, there are many plugins introducing their own syntax, and on the other hand, the sorting specification syntax is rich, and you can list files and folders names, it is asking for syntax conflicts.

Now, the issue which you bring up: unwanted entry in property names autosuggestion. I get it. Because the properties UI (and its autosuggestion) is a relatively new mechanism to Obsidian, let's wait until it matures. Maybe the Obsidian team will introduce a way to exclude certain properties from autosuggestion?

I'm introducing (soon) a fully config-less mode of custom sorting, via integration of bookmarks and bookmarks UI drag & drop feature to order items manually, indirectly. Maybe this will be suitable for your use case?

Maybe you have some ideas to address your concern and keep the plugin fundamental design intact? I'd like to avoid introducing a note-level syntax for sorting, maybe there are other equivalent options?

The introduction of properties UI is a game changer or a headache, depending on how a person uses Obsidian. Since then I'm thinking about an alternative way of editing the sorting spec, which wouldn't be an overkill at the same time.

**Comment by arteny** [comment]
@SebastianMC thank you very much for such a detailed answer. if you asking what is best way to use sorting for me, it is just to move items in file explorer, but not sure how it is easy or possible at all in implementation.
And I understood about property usage sense, only i found one lack here - it doesn't support integrated properties lists. 
What I mean:
```
---
sorting-spec: |-
  Item 1
  Item 2
---
```
supported

```
---
sorting-spec:
  - Item 1
  - Item 2
---
```
not supported

Also support of notes links might be useful (to avoid misprint errors and more faster preparing of sorting list)

**Comment by SebastianMC** [comment]
Hi @arteny 

as you see, I migrated this thread to discussions sections here.

I need to give some more thought to the idea of 'integrated properties lists', suspecting some syntax conflicts between YAML and valid file names and the sorting specification syntax.

Support for notes links is very tempting! Similarly to the idea of generating list of entries (folders and notes) in a folder via some context menu.

I parked the ideas on the roadmap.

At the same time you can take a look at the recently released drag and drop ordering of File Explorer items via Bookmarks integration. It is available in the recent [2.0.2 release of the plugin](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/2.0.2)

**Reply by arteny** [reply]
with sorting using bookmarks released, this my suggestions not much important for me more, new feature looks like more useful, thanks
